add-on/chrome/modules/MozLoopService.jsm

```diff
@@ -796,13 +795,16 @@ var MozLoopServiceInternal = {
   /**
    * Saves loop logs to the saved-telemetry-pings folder.
    *
-   * @param {Object} pc The peerConnection in question.
+   * @param {nsIDOMWindow} window The window object which can be communicated with
+   * @param {Object}        The peerConnection in question.
    */
-  stageForTelemetryUpload: function(window, pc) {
-    window.WebrtcGlobalInformation.getAllStats(allStats => {
-      let internalFormat = allStats.reports[0]; // filtered on pc.id
-      window.WebrtcGlobalInformation.getLogging("", logs => {
-        let report = convertToRTCStatsReport(internalFormat);
+  stageForTelemetryUpload: function(window, details) {
+    let mm = window.messageManager;
+    mm.addMessageListener("Loop:GetAllWebrtcStats", message => {
```

Comment:
I believe listeners on the message manager share the objects' lifetime, thus are cleaned up when the messageManager is destroyed.